pkg/logger/v2/logger.go

```diff
@@ -0,0 +1,169 @@
+package v2
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"runtime"
+	"strings"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/config/build"
+)
+
+type Config struct {
+	Name  string
+	Level slog.Leveler
+
+	// (optional) Logger helps convert existing zap.Logger to slog.Logger.
+	Logger *zap.Logger
+}
+
+func (c Config) New() *slog.Logger {
+	if c.Level == nil {
+		c.Level = slog.LevelDebug
+	}
+
+	if c.Logger == nil {
+		c.Logger = zap.L()
+	}
+
+	return slog.New(&zapHandler{
+		config: c,
+		attributes: []slog.Attr{
+			slog.String("version", buildVersion()),
+		},
+		groups: []string{},
+	})
+}
+
+func Named(name string, logger *slog.Logger) *slog.Logger {
+	if logger == nil {
+		return Config{
+			Name: name,
+		}.New()
+	}
+
+	handler := logger.Handler()
+	if hndlr, ok := handler.(ZapHandler); ok {
+		return slog.New(hndlr.WithName(name))
+	}
+
+	return logger
+}
+
+type ZapHandler interface {
+	WithName(string) slog.Handler
+}
```

Comment:
I think we could inline this and maybe keep it unexported until there is a need to expose it? I was worried more about multiple internal types than external.
```suggestion
	if hndlr, ok := handler.(interface { withName(string) slog.Handler }); ok {
		return slog.New(hndlr.WithName(name))
	}

	return logger
}
```

Reply:
That works. I had it this way to run some tests, but it's not necessary.